pkgs/servers/computing/slurm/default.nix

```diff
@@ -76,12 +91,52 @@ stdenv.mkDerivation rec {
 
 
   preConfigure = ''
-    patchShebangs ./doc/html/shtml2html.py
-    patchShebangs ./doc/man/man2html.py
+   # We're going to move the internal libraries that cause cyclic dependencies to $out,
+   # and only keep things like libpmi2 in $lib:
+   export NIX_LDFLAGS="-rpath ''${!outputLib}/lib/slurm -rpath ''${!outputBin}/lib/slurm ''${NIX_LDFLAGS-}"
+
+    while IFS= read -r -d ''$'\0' path ; do
+      patchShebangs "$path"
+    done < <( find -print0 -iname '*.py' -or -iname '*.sh')
   '';
 
-  postInstall = ''
-    rm -f $out/lib/*.la $out/lib/slurm/*.la
+  makeFlags = [
+    "perlpath=${lib.getExe buildPackages.perl}"
```

Comment:
perl in nativeBuildInputs is not sufficient?

Reply:
No, there's a hard-coded FHS path. Maybe I add a comment

Reply:
Added the comment